examples/machine_learning/geneticalgorithm.cpp

```diff
@@ -0,0 +1,202 @@
+/*******************************************************
+ * Copyright (c) 2014, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#include <arrayfire.h>
+#include <stdio.h>
+#include <string.h>
+#include <math.h>
+#include <limits.h>
+
+#include <iostream>
+
+#define OFFSET n/100
+
+using namespace af;
+
+array searchSpace_x_display = 0;
+array searchSpace_y_display = 0;
+array searchSpace_z_display = 0;
+array searchSpace;
+
+array samplex;
+array sampley;
+array samplez;
```

Comment:
Global are bad. The only reason to use them should be for input data for benchmarking with `timeit`.